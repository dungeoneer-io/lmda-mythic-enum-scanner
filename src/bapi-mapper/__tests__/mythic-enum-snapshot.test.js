const { describe, expect, test } = require('@jest/globals');

const {
    mapApiDungeonToPeriodDungeon,
    mapApiAffixToAffix,
    mapApiPeriodToPeriodWithoutDungeons
} = require('../mythic-enum-snapshot');

const ArrayOfPeriods = require('./__fixtures__/array-of-periods.json');
const AffixIndex = require('./__fixtures__/affix-index.json');
const ArrayOfDungeons = require('./__fixtures__/array-of-dungeons.json');

describe('transformers', () => {
    test('mapApiDungeonToPeriodDungeon', () => {
        const actual = ArrayOfDungeons.map(mapApiDungeonToPeriodDungeon);
        expect(actual).toMatchSnapshot();
    });

    test('mapApiAffixToAffix', () => {
        const result = AffixIndex.map(mapApiAffixToAffix);
        expect(result).toMatchSnapshot();
    });

    test('mapApiPeriodToPeriodWithoutDungeons', () => {
        const result = ArrayOfPeriods.map(mapApiPeriodToPeriodWithoutDungeons);
        expect(result).toMatchSnapshot();
    });
});