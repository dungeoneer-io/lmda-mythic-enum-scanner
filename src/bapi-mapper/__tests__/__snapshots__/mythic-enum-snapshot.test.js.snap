// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformers mapApiAffixToAffix 1`] = `
[
  {
    "id": 1,
    "name": "Overflowing",
  },
  {
    "id": 2,
    "name": "Skittish",
  },
  {
    "id": 3,
    "name": "Volcanic",
  },
  {
    "id": 4,
    "name": "Necrotic",
  },
  {
    "id": 5,
    "name": "Teeming",
  },
  {
    "id": 6,
    "name": "Raging",
  },
  {
    "id": 7,
    "name": "Bolstering",
  },
  {
    "id": 8,
    "name": "Sanguine",
  },
  {
    "id": 9,
    "name": "Tyrannical",
  },
  {
    "id": 10,
    "name": "Fortified",
  },
  {
    "id": 11,
    "name": "Bursting",
  },
  {
    "id": 12,
    "name": "Grievous",
  },
  {
    "id": 13,
    "name": "Explosive",
  },
  {
    "id": 14,
    "name": "Quaking",
  },
  {
    "id": 16,
    "name": "Infested",
  },
  {
    "id": 117,
    "name": "Reaping",
  },
  {
    "id": 119,
    "name": "Beguiling",
  },
  {
    "id": 120,
    "name": "Awakened",
  },
  {
    "id": 128,
    "name": "Tormented",
  },
  {
    "id": 121,
    "name": "Prideful",
  },
  {
    "id": 122,
    "name": "Inspiring",
  },
  {
    "id": 123,
    "name": "Spiteful",
  },
  {
    "id": 124,
    "name": "Storming",
  },
  {
    "id": 130,
    "name": "Encrypted",
  },
  {
    "id": 129,
    "name": "Infernal",
  },
  {
    "id": 131,
    "name": "Shrouded",
  },
]
`;

exports[`transformers mapApiDungeonToPeriodDungeon 1`] = `
[
  {
    "dungeon": 558,
    "id": 169,
    "map": 1195,
    "name": "Iron Docks",
    "tracked": true,
    "upgrades": [
      1800000,
      1440000,
      1080000,
    ],
    "zone": "iron-docks",
  },
  {
    "dungeon": 740,
    "id": 199,
    "map": 1501,
    "name": "Black Rook Hold",
    "tracked": false,
    "upgrades": [
      2340000,
      1872000,
      1404000,
    ],
    "zone": "black-rook-hold",
  },
  {
    "dungeon": 707,
    "id": 207,
    "map": 1493,
    "name": "Vault of the Wardens",
    "tracked": false,
    "upgrades": [
      1980000,
      1584000,
      1188000,
    ],
    "zone": "vault-of-the-wardens",
  },
]
`;

exports[`transformers mapApiPeriodToPeriodWithoutDungeons 1`] = `
[
  {
    "end_timestamp": 1661871599000,
    "id": 869,
    "start_timestamp": 1661266800000,
  },
  {
    "end_timestamp": 1662476399000,
    "id": 870,
    "start_timestamp": 1661871600000,
  },
]
`;
